src/main/java/seedu/address/model/interview/Interview.java

```diff
@@ -205,6 +206,19 @@ public void setStatus(InterviewStatus status) {
         this.status = status;
     }
 
+    /**
+     * Checks if another interview has timing overlaps with this interview.
+     * @param other the interview to check against.
+     * @return true if they start at the same time or if one begins before the other ends.
+     */
+    public boolean hasOverLapWith(Interview other) {
+        LocalDateTime start = LocalDateTime.of(getDate(), getStartTime());
+        LocalDateTime end = LocalDateTime.of(getDate(), getEndTime());
+        LocalDateTime toScheduleStart = LocalDateTime.of(other.getDate(), other.getStartTime());
+        return toScheduleStart.isAfter(start) && toScheduleStart.isBefore(end)
+                || start.isEqual(toScheduleStart);
+    }
+
```

Comment:
Let's say c1 already has i1 : 1300-1500

If assign c1 to i2: 1200-1400, wouldn't this be allowed? Cause 1200 !> 1300 and 1200<1400

Reply:
yep, just updated it for a bidirectional check

Reply:
oh sorry i misunderstood your comment, the getHour() checks are only for interviews that go beyond 1 "day", to update the variable with correct LocalDateTime coz we are not creating the "other" LocalDate object anywhere